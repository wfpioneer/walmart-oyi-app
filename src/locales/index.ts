import * as RNLocalize from 'react-native-localize';
import I18n from 'i18n-js';
// @ts-ignore
import moment from 'moment';
import 'moment/locale/es';
import en from './en';
import es from './es';
import zh from './zh';

export const setI18nConfig: () => void = () => {
  // fallback if no available language fits
  const fallback = { languageTag: 'en', isRTL: false };

  const { languageTag } = RNLocalize.findBestAvailableLanguage(['en', 'es', 'zh'])
    || fallback;

  // set i18n-js config
  I18n.translations = { en, es, zh };
  I18n.locale = languageTag;

  // the following is to set up moment to work with spanish localization
  moment.locale(languageTag);
};

export const setLanguage = (language: string) => {
  if (language !== 'en' && language !== 'es' && language !== 'zh') {
    throw new Error('Unsupported language');
  }

  I18n.locale = language;
  moment.locale(language);
};

export const strings: (
  tag: string,
  options?: I18n.TranslateOptions | undefined,
) => string = (tag: string, options: I18n.TranslateOptions | undefined = {}) => I18n.t(tag, options);

export const currencies: (
  value: number,
  options?: I18n.ToCurrencyOptions | undefined
) => string = (value: number, options: I18n.ToCurrencyOptions | undefined = {}) => I18n.toCurrency(value, options);

export const numbers: (
  value: number,
  options?: I18n.ToNumberOptions | undefined
) => string = (value: number, options: I18n.ToNumberOptions | undefined = {}) => I18n.toNumber(value, options);
