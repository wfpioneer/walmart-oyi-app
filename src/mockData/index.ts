
import itemDetails from './getItemDetails';


export const getMockItemDetails = (upc: string) => {
  // @ts-ignore
  return itemDetails[upc] || {};
};
