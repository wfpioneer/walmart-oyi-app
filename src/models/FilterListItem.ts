interface FilterListItem {
    value: string;
    display: string;
    selected: boolean;
  }

export default FilterListItem;
