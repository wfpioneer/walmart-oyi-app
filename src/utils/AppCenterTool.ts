import AppCenter from 'appcenter';
import AppCenterCrashes from 'appcenter-crashes';
import AppCenterAnalytics from 'appcenter-analytics';
import store from '../state';

export const initialize = (): void => {
  if (!__DEV__) {
    AppCenter.setLogLevel(AppCenter.LogLevel.VERBOSE);
    AppCenterCrashes.notifyUserConfirmation(AppCenterCrashes.UserConfirmation.ALWAYS_SEND);
    AppCenterCrashes.setEnabled(true);
    AppCenterAnalytics.setEnabled(true);
  }
};

export const trackEvent = (eventName: string, params: any = {}): void => {
  const userState = store.getState().User;
  // Removes undefined values from the event properties
  const cleanedParams = Object.fromEntries(Object.entries(params).filter(([k, v]) => v !== undefined));
  const submitParams = {
    ...cleanedParams,
    userId: userState.userId,
    clubId: userState.siteId.toString(),
    country: userState.countryCode
  };
  AppCenterAnalytics.trackEvent(eventName, submitParams);
};

export const setUserId = (userId: string): void => {
  AppCenter.setUserId(userId);
};
