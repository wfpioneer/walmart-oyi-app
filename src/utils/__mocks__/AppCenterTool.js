export default {
  initialize: jest.fn(),
  trackEvent: jest.fn(() => Promise.resolve()),
  setUserId: jest.fn(() => Promise.resolve())
};
