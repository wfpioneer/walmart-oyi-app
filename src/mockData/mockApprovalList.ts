import { ApprovalListItem, approvalRequestSource, approvalStatus } from '../models/ApprovalListItem';

enum CATEGORY_DESCRIPTION {
  SNACKS = 'SNACKS',
  FRUITS_VEGETABLES = 'FRUITS/VEGETABLES',
}

enum INITIATED_USER_ID {
  ASSOCIATE_EMPLOYEE = 'Associate Employee',
  ASSOCIATE_WORKER = 'Associate Worker',
}

enum INITIATED_TIMESTAMP {
  DAY1 = '2021-03-27T00:00:00.000Z',
  DAY2 = '2021-03-28T00:00:00.000Z'
}

export const mockApprovals: ApprovalListItem[] = [{
  imageUrl: undefined,
  itemName: 'Nature Valley Crunchy Cereal Bars ',
  itemNbr: 123,
  upcNbr: 40000000123,
  categoryNbr: 1,
  categoryDescription: CATEGORY_DESCRIPTION.SNACKS,
  subCategoryNbr: 1,
  subCategoryDescription: '',
  newQuantity: 20,
  oldQuantity: 5,
  dollarChange: 150.50,
  initiatedUserId: INITIATED_USER_ID.ASSOCIATE_EMPLOYEE,
  initiatedTimestamp: INITIATED_TIMESTAMP.DAY1,
  approvalStatus: approvalStatus.Pending,
  approvalRequestSource: approvalRequestSource.ItemDetails,
  isChecked: false,
  daysLeft: 3
}, {
  imageUrl: undefined,
  itemName: 'Red Apples',
  itemNbr: 456,
  upcNbr: 40000000456,
  categoryNbr: 2,
  categoryDescription: CATEGORY_DESCRIPTION.FRUITS_VEGETABLES,
  subCategoryNbr: 1,
  subCategoryDescription: '',
  newQuantity: 10,
  oldQuantity: 30,
  dollarChange: 50.00,
  initiatedUserId: 'Associate Worker',
  initiatedTimestamp: INITIATED_TIMESTAMP.DAY2,
  approvalStatus: approvalStatus.Pending,
  approvalRequestSource: approvalRequestSource.ItemDetails,
  isChecked: false,
  daysLeft: 2
}, {
  imageUrl: undefined,
  itemName: 'Cabbage',
  itemNbr: 457,
  upcNbr: 40000000457,
  categoryNbr: 2,
  categoryDescription: CATEGORY_DESCRIPTION.FRUITS_VEGETABLES,
  subCategoryNbr: 1,
  subCategoryDescription: '',
  newQuantity: 15,
  oldQuantity: 30,
  dollarChange: 40.00,
  initiatedUserId: INITIATED_USER_ID.ASSOCIATE_WORKER,
  initiatedTimestamp: INITIATED_TIMESTAMP.DAY2,
  approvalStatus: approvalStatus.Pending,
  approvalRequestSource: approvalRequestSource.ItemDetails,
  isChecked: false,
  daysLeft: 2
}, {
  imageUrl: undefined,
  itemName: 'Strawberries',
  itemNbr: 458,
  upcNbr: 40000000458,
  categoryNbr: 2,
  categoryDescription: CATEGORY_DESCRIPTION.FRUITS_VEGETABLES,
  subCategoryNbr: 1,
  subCategoryDescription: '',
  newQuantity: 40,
  oldQuantity: 20,
  dollarChange: 80.00,
  initiatedUserId: INITIATED_USER_ID.ASSOCIATE_WORKER,
  initiatedTimestamp: INITIATED_TIMESTAMP.DAY2,
  approvalStatus: approvalStatus.Pending,
  approvalRequestSource: approvalRequestSource.ItemDetails,
  isChecked: false,
  daysLeft: 2
}];

export const mockSelectedApprovals: ApprovalListItem[] = [{
  imageUrl: undefined,
  itemName: 'Nature Valley Crunchy Cereal Bars ',
  itemNbr: 123,
  upcNbr: 40000000123,
  categoryNbr: 1,
  categoryDescription: CATEGORY_DESCRIPTION.SNACKS,
  subCategoryNbr: 1,
  subCategoryDescription: '',
  newQuantity: 20,
  oldQuantity: 5,
  dollarChange: 150.50,
  initiatedUserId: INITIATED_USER_ID.ASSOCIATE_EMPLOYEE,
  initiatedTimestamp: INITIATED_TIMESTAMP.DAY1,
  approvalStatus: approvalStatus.Pending,
  approvalRequestSource: approvalRequestSource.ItemDetails,
  isChecked: true
}, {
  imageUrl: undefined,
  itemName: 'Red Apples',
  itemNbr: 456,
  upcNbr: 40000000456,
  categoryNbr: 2,
  categoryDescription: CATEGORY_DESCRIPTION.FRUITS_VEGETABLES,
  subCategoryNbr: 1,
  subCategoryDescription: '',
  newQuantity: 10,
  oldQuantity: 30,
  dollarChange: 50.00,
  initiatedUserId: INITIATED_USER_ID.ASSOCIATE_WORKER,
  initiatedTimestamp: INITIATED_TIMESTAMP.DAY1,
  approvalStatus: approvalStatus.Pending,
  approvalRequestSource: approvalRequestSource.ItemDetails,
  isChecked: true
}, {
  imageUrl: undefined,
  itemName: 'Cabbage',
  itemNbr: 457,
  upcNbr: 40000000457,
  categoryNbr: 2,
  categoryDescription: CATEGORY_DESCRIPTION.FRUITS_VEGETABLES,
  subCategoryNbr: 1,
  subCategoryDescription: '',
  newQuantity: 15,
  oldQuantity: 30,
  dollarChange: 40.00,
  initiatedUserId: INITIATED_USER_ID.ASSOCIATE_WORKER,
  initiatedTimestamp: INITIATED_TIMESTAMP.DAY2,
  approvalStatus: approvalStatus.Pending,
  approvalRequestSource: approvalRequestSource.ItemDetails,
  isChecked: true
}, {
  imageUrl: undefined,
  itemName: 'Strawberries',
  itemNbr: 458,
  upcNbr: 40000000458,
  categoryNbr: 2,
  categoryDescription: CATEGORY_DESCRIPTION.FRUITS_VEGETABLES,
  subCategoryNbr: 1,
  subCategoryDescription: '',
  newQuantity: 40,
  oldQuantity: 20,
  dollarChange: 80.00,
  initiatedUserId: INITIATED_USER_ID.ASSOCIATE_WORKER,
  initiatedTimestamp: INITIATED_TIMESTAMP.DAY2,
  approvalStatus: approvalStatus.Pending,
  approvalRequestSource: approvalRequestSource.ItemDetails,
  isChecked: true
}];
