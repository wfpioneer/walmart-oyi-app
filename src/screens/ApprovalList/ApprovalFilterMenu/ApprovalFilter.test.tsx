import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import { fireEvent, render } from '@testing-library/react-native';
import store from '../../../state';
import {
  ApprovalFilterScreen,
  RenderSourceCollapsibleCard,
  renderSourceFilterCard
} from './ApprovalFilter';
import { mockApprovals } from '../../../mockData/mockApprovalList';
import { strings } from '../../../locales';
import { trackEvent } from '../../../utils/AppCenterTool';
import { mockSourceMap } from '../../../mockData/mockWorkList';
import { FilteredCategory } from '../../../models/FilterListItem';

jest.mock('../../../utils/AppCenterTool.ts', () => ({
  ...jest.requireActual('../../../utils/__mocks__/AppCenterTool'),
  trackEvent: jest.fn()
}));
jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');
jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon');

describe('Approval filter render tests', () => {
  const mockDispatch = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('ShallowRenders the filter screen', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <Provider store={store}>
        <ApprovalFilterScreen
          dispatch={mockDispatch}
          approvalList={mockApprovals}
          categoryOpen={false}
          filteredCategories={[]}
          sourceOpen={false}
          filteredSources={[]}
        />
      </Provider>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Test renders the filter menu component and calls onPress Function', async () => {
    const { findByText, toJSON } = render(
      <Provider store={store}>
        <ApprovalFilterScreen
          dispatch={mockDispatch}
          categoryOpen={false}
          filteredCategories={[]}
          sourceOpen={false}
          filteredSources={[]}
          approvalList={mockApprovals}
        />
      </Provider>
    );
    // You can query string translations
    const clearButton = findByText(strings('WORKLIST.CLEAR'));
    fireEvent.press(await clearButton);
    expect(trackEvent).toBeCalledTimes(1);
    expect(mockDispatch).toBeCalledTimes(1);
    expect(toJSON()).toMatchSnapshot();
  });
});

describe('Approval Filter Externalized Function Tests', () => {
  const mockUpdateFilterSources = jest.fn();
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Tests renderSourceFilterCard and calls onPress event', async () => {
    // Audit Filter Request Source Selected
    const { findByTestId, update } = render(
      renderSourceFilterCard(mockSourceMap[0], [], mockUpdateFilterSources)
    );
    const itemPress = findByTestId('category button');
    fireEvent.press(await itemPress);
    expect(mockUpdateFilterSources).toBeCalledWith([
      mockSourceMap[0].catgName
    ]);

    // Item Details Filter Request Source Selected
    update(
      renderSourceFilterCard(
        mockSourceMap[1],
        [mockSourceMap[0].catgName],
        mockUpdateFilterSources
      )
    );
    fireEvent.press(await itemPress);
    expect(mockUpdateFilterSources).toBeCalledWith([
      mockSourceMap[0].catgName,
      mockSourceMap[1].catgName,
      ''
    ]);

    // TODO Remove when no approval items lack a source
    const noCatg: FilteredCategory = {
      catgName: '',
      selected: false
    };
    const renderSourceNoElement = renderSourceFilterCard(
      noCatg,
      [],
      mockUpdateFilterSources
    );
    expect(renderSourceNoElement).toStrictEqual(<></>);
  });

  it('Tests renderSourceFilterCard and calls onPress event de-selecting source filters', async () => {
    const allSourcesFiltered = [
      mockSourceMap[0].catgName,
      mockSourceMap[1].catgName,
      ''
    ];
    mockSourceMap[0].selected = true;
    mockSourceMap[1].selected = true;

    // Audit Filter Request Source cleared Selection
    const { findByTestId, update } = render(
      renderSourceFilterCard(
        mockSourceMap[0],
        allSourcesFiltered,
        mockUpdateFilterSources
      )
    );
    const itemPress = findByTestId('category button');
    fireEvent.press(await itemPress);
    const removedAuditSource = [mockSourceMap[1].catgName, ''];
    expect(trackEvent).toBeCalledWith('approvals_update_filter_source', {
      categories: JSON.stringify(removedAuditSource)
    });
    expect(mockUpdateFilterSources).toBeCalledWith(removedAuditSource);

    // Item Details Filter Request Source Cleared Selection
    update(
      renderSourceFilterCard(
        mockSourceMap[1],
        removedAuditSource,
        mockUpdateFilterSources
      )
    );
    fireEvent.press(await itemPress);
    expect(trackEvent).toBeCalledWith('approvals_update_filter_source', {
      categories: JSON.stringify([])
    });
    expect(mockUpdateFilterSources).toBeCalledWith([]);
  });

  it('Tests RenderSourceCollapsibleCard and calls toggle source menu onPress event', async () => {
    const mockToggleSrcs = jest.fn();
    const { findByTestId, update } = render(
      <RenderSourceCollapsibleCard
        sourceMap={mockSourceMap}
        sourceOpen={true}
        filterSources={['audits']}
        toggleSrcs={mockToggleSrcs}
        updateFilterSrcs={mockUpdateFilterSources}
      />
    );

    const toggleSrcButton = findByTestId('toggle sources');
    fireEvent.press(await toggleSrcButton);
    expect(mockToggleSrcs).toBeCalledWith(false);

    // Source open set to false
    update(
      <RenderSourceCollapsibleCard
        sourceMap={mockSourceMap}
        sourceOpen={false}
        filterSources={['audits']}
        toggleSrcs={mockToggleSrcs}
        updateFilterSrcs={mockUpdateFilterSources}
      />
    );
    fireEvent.press(await toggleSrcButton);
    expect(mockToggleSrcs).toBeCalledWith(true);
  });
});
