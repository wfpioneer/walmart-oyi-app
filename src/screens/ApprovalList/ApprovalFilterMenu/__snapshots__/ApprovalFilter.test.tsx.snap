// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Approval filter render tests ShallowRenders the filter screen 1`] = `
<Context.Provider
  value={
    {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ApprovalFilterScreen
    approvalList={
      [
        {
          "approvalRequestSource": "itemdetails",
          "approvalStatus": "Pending",
          "categoryDescription": "SNACKS",
          "categoryNbr": 1,
          "daysLeft": 3,
          "dollarChange": 150.5,
          "imageUrl": undefined,
          "initiatedTimestamp": "2021-03-27T00:00:00.000Z",
          "initiatedUserId": "Associate Employee",
          "isChecked": false,
          "itemName": "Nature Valley Crunchy Cereal Bars ",
          "itemNbr": 123,
          "newQuantity": 20,
          "oldQuantity": 5,
          "subCategoryDescription": "",
          "subCategoryNbr": 1,
          "upcNbr": 40000000123,
        },
        {
          "approvalRequestSource": "itemdetails",
          "approvalStatus": "Pending",
          "categoryDescription": "FRUITS/VEGETABLES",
          "categoryNbr": 2,
          "daysLeft": 2,
          "dollarChange": 50,
          "imageUrl": undefined,
          "initiatedTimestamp": "2021-03-28T00:00:00.000Z",
          "initiatedUserId": "Associate Worker",
          "isChecked": false,
          "itemName": "Red Apples",
          "itemNbr": 456,
          "newQuantity": 10,
          "oldQuantity": 30,
          "subCategoryDescription": "",
          "subCategoryNbr": 1,
          "upcNbr": 40000000456,
        },
        {
          "approvalRequestSource": "itemdetails",
          "approvalStatus": "Pending",
          "categoryDescription": "FRUITS/VEGETABLES",
          "categoryNbr": 2,
          "daysLeft": 2,
          "dollarChange": 40,
          "imageUrl": undefined,
          "initiatedTimestamp": "2021-03-28T00:00:00.000Z",
          "initiatedUserId": "Associate Worker",
          "isChecked": false,
          "itemName": "Cabbage",
          "itemNbr": 457,
          "newQuantity": 15,
          "oldQuantity": 30,
          "subCategoryDescription": "",
          "subCategoryNbr": 1,
          "upcNbr": 40000000457,
        },
        {
          "approvalRequestSource": "audits",
          "approvalStatus": "Pending",
          "categoryDescription": "FRUITS/VEGETABLES",
          "categoryNbr": 2,
          "daysLeft": 2,
          "dollarChange": 80,
          "imageUrl": undefined,
          "initiatedTimestamp": "2021-03-28T00:00:00.000Z",
          "initiatedUserId": "Associate Worker",
          "isChecked": false,
          "itemName": "Strawberries",
          "itemNbr": 458,
          "newQuantity": 40,
          "oldQuantity": 20,
          "subCategoryDescription": "",
          "subCategoryNbr": 1,
          "upcNbr": 40000000458,
        },
      ]
    }
    categoryOpen={false}
    dispatch={[MockFunction]}
    filteredCategories={[]}
    filteredSources={[]}
    sourceOpen={false}
  />
</Context.Provider>
`;

exports[`Approval filter render tests Test renders the filter menu component and calls onPress Function 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#0071e9",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "flex-start",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffffff",
          "flex": 1,
          "marginLeft": 15,
        }
      }
    >
      [missing "en.WORKLIST.REFINE" translation]
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 0.5,
          "marginRight": 15,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffffff",
            "textAlign": "right",
          }
        }
      >
        [missing "en.WORKLIST.CLEAR" translation]
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#e0e0e0",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 15,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text>
        [missing "en.WORKLIST.CATEGORY" translation]
      </Text>
      <Text
        style={
          {
            "color": "#9e9e9e",
            "fontSize": 12,
          }
        }
      >
        [missing "en.WORKLIST.ALL" translation]
      </Text>
    </View>
    <View
      style={
        {
          "flex": 0.1,
        }
      }
    >
      <Icon
        color="#616161"
        name="keyboard-arrow-down"
        size={25}
      />
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#e0e0e0",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 15,
      }
    }
    testID="toggle sources"
  >
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text>
        [missing "en.APPROVAL.SOURCE" translation]
      </Text>
      <Text
        style={
          {
            "color": "#9e9e9e",
            "fontSize": 12,
          }
        }
      >
        [missing "en.WORKLIST.ALL" translation]
      </Text>
    </View>
    <View
      style={
        {
          "flex": 0.1,
        }
      }
    >
      <Icon
        color="#616161"
        name="keyboard-arrow-down"
        size={25}
      />
    </View>
  </View>
</View>
`;
